### DISCUSSION TITLE
Block individual paragraph if it contains a certain word, without blocking other paragraphs in the same HTML tag.

### DISCUSSION THREAD

**Discussion by ItsIgnacioPortal** [uBlockOrigin/uBlock-issues#2377] [Q&A] [answered]
Given a page like this:
```html
<tbody>
    <tr>
        <td class="mcnTextContent" valign="top">
            Almost 50% of proof of stake nodes for Ethereum were attributed to just two people recently. Hopefully that will open up soon, but there's no guarantee that it will. We should all watch closely when something designed to be decentralized ends up consolidating into just a few hands. <span><span><a href="REDACTED">MORE</span></span></a><br>
            <br>
            Buy our product USELESS-TRASH now. <span><span><a href="REDACTED">MORE</span></span></a><br>
            <br>
            Google has canceled half of its R&amp;D projects at Area 120, its internal R&amp;D group. This is part of Google's general push toward waste-cutting.&nbsp;<span><span><a href="REDACTED">MORE</a></span></span><br>
            <br>
        </td>
    </tr>
</tbody>
```

How could I block the paragraph that contains the term `USELESS-TRASH`, without blocking the sorrounding paragraphs? Note that the ad isn't always in the second paragraph, sometimes it's in the first paragraph, and sometimes it's in the last paragraph. Every filter I've tried just blocks the entire `td` element :/

**Comment by ghost** [comment]
Not possible.

**Comment by krystian3w** [comment]
Needed userscript or custom scriptlet to modify dom/node.

**Reply by krystian3w** [reply]
@ItsIgnacioPortal you can test these: https://github.com/uBlockOrigin/uAssets/discussions/18185 in dev build.

For sure, the most difficult thing will be to write so that the filter works optimally on the site (for node hooks you have only `td` or `#text`) and does not generate a message that the add-on overloads the browser.

Link should be hidden in procedural filtering if can escape `[href*="xyz" i]`.

**Comment by gorhill** [comment, marked as answer]
Isn't this a case for the new scriptlet `remove-node-text`?

    example.com##+js(rmnt, #text, USELESS-TRASH)
    example.com##.mcnTextContent br + span > span > a

**Reply by garry-ut99** [reply]
- https://github.com/uBlockOrigin/uBlock-issues/issues/1654 +
- https://github.com/uBlockOrigin/uBlock-issues/issues/1777 ?

**Reply by ItsIgnacioPortal** [reply]
Yes! That's perfect! thank you so much :D

---

Context for the unaware: In order to try this you need to download a dev version of uBlockOrigin: https://github.com/gorhill/uBlock/releases
You also can't just install this over the regular uBO, you have to uninstall it and then install this version. If you don't want to do that (understandably), you can go to (about:profiles)[about:profiles] and test it in an empty browser profile.

**Reply by gorhill** [reply]
> You also can't just install this over the regular uBO, you have to uninstall it and then install this version.

That's only true for Chromium, and you don't need to uninstall the stable release, just disable it.

**Reply by krystian3w** [reply]
Once downgrade uBo dev to stable broke cosmetics filtering in Firefox, so better backup settings.

**Reply by gorhill** [reply]
> broke cosmetics filtering in Firefox

Probably just need to force filter lists to be re-compiled.
